pkg/standalone-utils/contracts/BalancerFeeBurner.sol

```diff
@@ -0,0 +1,133 @@
+        }
+    }
+
+    /// @inheritdoc IProtocolFeeBurner
+    function burn(
+        address pool,
+        IERC20 feeToken,
+        uint256 feeTokenAmount,
+        IERC20 targetToken,
+        uint256 minAmountOut,
+        address recipient,
+        uint256 deadline
+    ) external authenticate {
+        getVault().unlock(
+            abi.encodeCall(
+                BalancerFeeBurner.burnHook,
+                BurnHookParams({
+                    pool: pool,
+                    sender: msg.sender,
+                    feeToken: feeToken,
+                    feeTokenAmount: feeTokenAmount,
+                    targetToken: targetToken,
+                    minAmountOut: minAmountOut,
+                    recipient: recipient,
+                    deadline: deadline
+                })
+            )
+        );
+    }
+
+    function burnHook(BurnHookParams calldata params) external nonReentrant onlyVault {
+        // The deadline is timestamp-based: it should not be relied upon for sub-minute accuracy.
+        // solhint-disable-next-line not-rely-on-time
+        if (block.timestamp > params.deadline) {
+            revert SwapDeadline();
+        }
+
+        IERC20 feeToken = params.feeToken;
+        IERC20 targetToken = params.targetToken;
+        uint256 feeTokenAmount = params.feeTokenAmount;
+
+        SwapPathStep[] memory steps = getBurnPath(feeToken);
+        uint256 lastStepIndex = steps.length - 1;
+        if (steps[lastStepIndex].tokenOut != targetToken) {
+            revert TargetTokenInPathNotTheSame();
+        }
+
+        IVault vault = getVault();
+
+        // Transfer the token in to the vault.
+        feeToken.safeTransferFrom(params.sender, address(vault), feeTokenAmount);
+        vault.settle(feeToken, feeTokenAmount);
+
+        // Swap the fee token for the target token through the steps.
+        IERC20 stepTokenIn = feeToken;
+        uint256 stepExactAmountIn = feeTokenAmount;
+        for (uint256 i = 0; i < steps.length; i++) {
+            SwapPathStep memory step = steps[i];
+
+            (, , uint256 amountOut) = vault.swap(
+                VaultSwapParams({
+                    kind: SwapKind.EXACT_IN,
+                    pool: step.pool,
+                    tokenIn: stepTokenIn,
+                    tokenOut: step.tokenOut,
+                    amountGivenRaw: stepExactAmountIn,
+                    limitRaw: (i == lastStepIndex) ? params.minAmountOut : 0,
+                    userData: new bytes(0)
+                })
+            );
+
+            stepTokenIn = step.tokenOut;
+            stepExactAmountIn = amountOut;
+        }
+
+        address recipient = params.recipient;
+        // Last stepTokenIn is the final token out. Last stepExactAmountIn is the amount out.
+        vault.sendTo(stepTokenIn, recipient, stepExactAmountIn);
+
+        emit ProtocolFeeBurned(params.pool, feeToken, feeTokenAmount, targetToken, stepExactAmountIn, recipient);
```

Comment:
```suggestion
        // Last stepTokenIn is the final token out. Last stepExactAmountIn is the amount out.
        vault.sendTo(stepTokenIn, params.recipient, stepExactAmountIn);

        emit ProtocolFeeBurned(params.pool, feeToken, feeTokenAmount, targetToken, stepExactAmountIn, params.recipient);
```
Does it save anything to copy the recipient here? Should be the same (and simpler) to just directly reference it from params.